docker_build.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+MY_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+# Echo you can override envronment variables to build images you want
+# PYTHON_VERSION [2.7, 3.5, 3.6] - you can choose which python version should be installed
+# DOCKERHUB_USER = defaults to airflow but you can override it to build images you want to
+#                  publish under different dockerhub user
+# PUSH_IMAGES = defaults to false - no images are pushed, but you can set it to true to push images to
+#               DockerHub. Useful when you use your own DOCKERHUB_USER and push images for caching for your
+#               organization
+set -euo pipefail
+
+set -x
+export PYTHON_VERSION=${PYTHON_VERSION:=$(python -c 'import sys; print("%s.%s" % (sys.version_info.major, sys.version_info.minor))')}
+export DOCKERHUB_USER=${DOCKERHUB_USER:=airflow}
+export IMAGE_NAME=${DOCKERHUB_USER}/airflow:latest-${PYTHON_VERSION}
+export PUSH_IMAGES=${PUSH_IMAGES:=false}
```

Comment:
Just so you know: we can't push to dockerhub under apache - only we don't have permissions. I think we got the Infra team to set up docker auto-build for us.

I don't know if this changes any of your plans. Also: we should check this as I might be making things up.

Reply:
I dunderstand that, this is perfectly fine. What will eventually happen (if what I do will prove working well) is that DockerHub builds will auto-build and push the images from master only. Those images will be used by TravisCI as cache for builds (it will only eventually rebuild latest layer with sources locally + occasionally it will only install dependencies when they change. But this is something i am currently looking at to optimize a bit.

Reply:
I described how it works in https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-10+Multi-layered+and+multi-stage+official+Airflow+image, including the workflows and frequency of builds, when the images are pushed (only in master DockerHub). I will resolve the conversation now. Please reopen if you think we need to discuss more :).